espresso/batch_buffer.go

```diff
@@ -0,0 +1,79 @@
+package espresso
+
+import (
+	"cmp"
+	"slices"
+
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+type BatchValidity uint8
+
+const (
+	// BatchDrop indicates that the batch is invalid, and will always be in the future, unless we reorg
+	BatchDrop = iota
+	// BatchAccept indicates that the batch is valid and should be processed
+	BatchAccept
+	// BatchUndecided indicates we are lacking L1 information until we can proceed batch filtering
+	BatchUndecided
+	// BatchFuture indicates that the batch may be valid, but cannot be processed yet and should be checked again later
+	BatchFuture
+	// BatchPast indicates that the batch is from the past, i.e. its timestamp is smaller or equal
+	// to the safe head's timestamp.
+	BatchPast
+)
+
+type Batch interface {
+	Number() uint64
+	L1Origin() eth.BlockID
+	Hash() common.Hash
```

Comment:
Artifact of a workaround, can be removed

Reply:
I didn't remove this yet, because according to https://github.com/EspressoSystems/optimism-espresso-integration/pull/96#discussion_r2040253533, it seems that we'll need some refactoring after removing this.

Reply:
Oh, I meant Hash() function, it's not necessary for this interface.